assignments/lambda-classes.js

```diff
@@ -1 +1,112 @@
 // CODE here for your Lambda Classes
+
+class Person { 
+    constructor(baseclass) {
+        this.name = baseclass.name,
+        this.age = baseclass.age,
+        this.location = baseclass.location;
+    }
+    speak () {
+        return `Hello my name is ${this.name}, I am from ${this.location}.`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(teacher) {
+        super(teacher);
+        this.speciality = teacher.speciality,
+        this.favLanguage = teacher.favLanguage,
+        this.catchPhrase = teacher.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(student, subject) {
+       return `${student.name} receives a perfect score on ${subject}`;
+    }
+}
+
+class Student extends Person {
+    constructor(scholar) { 
+        super(scholar);
+        this.previousBackground = scholar.previousBackground,
+        this.className = scholar.className,
+        this.favSubjects = scholar.favSubjects,
+        this.grade = scholar.grade; //stretch challenge, trying to figure this out
+    }
+    listsSubjects () { //Im not sure if this is the correct way of doing this.
+       return `${this.favSubjects}`;
+    }
+    PRAssignment (student, subject) {
+      return `${student.name} has submitted a PR for ${subject}`;
+    }
+    sprintChallenge (student, subject) {
+      return `${student.name} has begun sprint challenge on ${subject}`;
+    }
+    grade () 
+  //stretch, trying to figure this out
+}
+
+class TeamLeads extends Instructor {
+    constructor(leaders) {
+        super(leaders);
+        this.gradClassName = leaders.gradClassName,
+        this.favInstructor = leaders.favInstructor; 
+    }
+    standUp (channel) {
+    return  `${this.name} announces to ${channel}, @channel standy times!​​​​​`;
+       // Why doesn't this work with just return ${name}? I am only wondering since this is how it is written in the instructions? 
+       // it will always be the name of the insctuctor.
+    }
```

Comment:
if you were passing in name as a parameter for standUp such as: `standUp(name, channel)` then you could just use `${name}` otherwise it will return name is undefined, but then when you call the standUp method you would have to also pass in whichever name you want doing the announcement. 

this.name is stating that this teamlead announces to the channel that it is standup time.